sw/device/tests/sim_dv/lc_ctrl_transition_test.c

```diff
@@ -21,6 +21,21 @@ static dif_lc_ctrl_t lc;
 
 const test_config_t kTestConfig;
 
+/**
+ * Track number of transactions under this test.
+ *
+ * From the software / compiler's perspective, this is a constant (hence the
+ * `const` qualifier). However, the external DV testbench finds this symbol's
+ * address and modifies it via backdoor, to track how many transactions have
+ * been sent. Hence, we add the `volatile` keyword to prevent the compiler from
+ * optimizing it out.
+ * The `const` is needed to put it in the .rodata section, otherwise it gets
+ * placed in .data section in the main SRAM. We cannot backdoor write anything
+ * in SRAM at the start of the test because the CRT init code wipes it to 0s.
+ */
+static volatile const uint8_t kTestIdx = 0x0;
+static volatile const uint8_t kTestNum = 0x0;
+
 // TODO: Issue more resets and andomize this array in each reset.
```

Comment:
Aah yes, issuing reset and repeating the test several times was precisely to address this TODO.